crates/loam-cli/src/commands/dev/mod.rs

```diff
@@ -0,0 +1,103 @@
+use clap::Parser;
+use notify::{RecursiveMode, Watcher};
+use std::path::{Path, PathBuf};
+use tokio::time;
+
+use crate::commands::build;
+
+use super::build::build_clients::LoamEnv;
+
+#[derive(Parser, Debug, Clone)]
+#[group(skip)]
+pub struct Cmd {
+    #[command(flatten)]
+    pub build_cmd: build::Cmd,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Watcher(#[from] notify::Error),
+    #[error(transparent)]
+    Build(#[from] build::Error),
+    #[error("IO error: {0}")]
+    Io(#[from] std::io::Error),
+}
+
+impl Cmd {
+    pub async fn run(&self) -> Result<(), Error> {
+        // Set LOAM_ENV to development
+        std::env::set_var("LOAM_ENV", "development");
+        let (tx, mut rx) = tokio::sync::mpsc::channel(100);
+        let tx_clone = tx.clone();
+        let mut watcher =
+            notify::recommended_watcher(move |res: Result<notify::Event, notify::Error>| {
+                if let Ok(event) = res {
```

Comment:
```suggestion
                if let Ok(event @ notify::Event { kind: notify::EventKind::Modify(_) } ) = res {
```
Since the assignment is a pattern match you can match on the inner type and the `@` lets you simultaneously assign the value to a variable. So this has the same effect as the nested ifs. See example here:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a2ae21a539c81d4d157644d6da6b978b

Reply:
Also this suggestion is not good on it's own since it doesn't remove the inner if.

Reply:
Also since these types don't collide with any in this file I'd just import them for better readability.

Reply:
done